packages/web3-core/src/web3_batch_request.ts

```diff
@@ -48,13 +48,15 @@ export class Web3BatchRequest {
 	}
 
 	// eslint-disable-next-line class-methods-use-this
-	public async execute(): Promise<JsonRpcBatchResponse<unknown, unknown>> {
+	public async execute(options?: {
+		timeout?: number;
+	}): Promise<JsonRpcBatchResponse<unknown, unknown>> {
 		if (this.requests.length === 0) {
 			return Promise.resolve([]);
 		}
 
 		const request = new Web3DeferredPromise<JsonRpcBatchResponse<unknown, unknown>>({
-			timeout: DEFAULT_BATCH_REQUEST_TIMEOUT,
+			timeout: options?.timeout ?? DEFAULT_BATCH_REQUEST_TIMEOUT,
```

Comment:
thanks for your contribution. This should be mentioned in changelog.

Reply:
apologies @jdevcs are you suggesting any change?

Reply:
@jdevcs Was the checklist not necessary? I couldn't even build the project and therefore abandoned the task.

Reply:
> apologies @jdevcs are you suggesting any change?

This should be added in changelog files : https://github.com/shubhamshd/web3.js/blob/4.x/packages/web3-core/CHANGELOG.md and https://github.com/shubhamshd/web3.js/blob/4.x/packages/web3/CHANGELOG.md

Reply:
> > @jdevcs Was the checklist not necessary? I couldn't even build the project and therefore abandoned the task.
> 
> Could you share problems you were stuck? Most of items are also checks in CI ( lint, build, tests etc)

Windows 11, node 18.17.0, yarn 1.22.19
<details><summary>Details</summary>
<p>

```
PS D:\Projects\web3.js> yarn install
yarn install v1.22.19
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
[4/5] Linking dependencies...
warning " > @cypress/webpack-preprocessor@5.12.2" has unmet peer dependency "@babel/core@^7.0.1".
warning " > @cypress/webpack-preprocessor@5.12.2" has unmet peer dependency "@babel/preset-env@^7.0.0".
warning " > @cypress/webpack-preprocessor@5.12.2" has unmet peer dependency "babel-loader@^8.0.2".
warning " > @droppedcode/typedoc-plugin-relative-includes@1.0.2" has incorrect peer dependency "typedoc@^0.22.11".
warning " > eslint-import-resolver-typescript@3.5.5" has unmet peer dependency "eslint-plugin-import@*".
warning "workspace-aggregator-efa93b05-b4d8-4200-a675-21919f1d0f7c > web3 > @truffle/hdwallet-provider > web3-provider-engine > eth-block-tracker > @babel/plugin-transform-runtime@7.18.10" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "workspace-aggregator-efa93b05-b4d8-4200-a675-21919f1d0f7c > web3 > @truffle/hdwallet-provider > web3-provider-engine > eth-block-tracker > @babel/plugin-transform-runtime > babel-plugin-polyfill-corejs2@0.3.2" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "workspace-aggregator-efa93b05-b4d8-4200-a675-21919f1d0f7c > web3 > @truffle/hdwallet-provider > web3-provider-engine > eth-block-tracker > @babel/plugin-transform-runtime > babel-plugin-polyfill-corejs3@0.5.3" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "workspace-aggregator-efa93b05-b4d8-4200-a675-21919f1d0f7c > web3 > @truffle/hdwallet-provider > web3-provider-engine > eth-block-tracker > @babel/plugin-transform-runtime > babel-plugin-polyfill-regenerator@0.4.0" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "workspace-aggregator-efa93b05-b4d8-4200-a675-21919f1d0f7c > web3 > @truffle/hdwallet-provider > web3-provider-engine > eth-block-tracker > @babel/plugin-transform-runtime > babel-plugin-polyfill-corejs2 > @babel/helper-define-polyfill-provider@0.3.2" has unmet peer dependency "@babel/core@^7.4.0-0".
[5/5] Building fresh packages...
$ yarn build
yarn run v1.22.19
$ yarn clean && ts-node scripts/init.ts
error An unexpected error occurred: "Cannot create property 'npmrc-tmpl' on string '{\"npmrc-tmpl\":true}'".
info If you think this is a bug, please open a bug report with the information provided in "D:\\Projects\\web3.js\\yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command
```

```
PS D:\Projects\web3.js> yarn build:esm 
yarn run v1.22.19
$ lerna run build:esm --stream
lerna notice cli v6.6.2
lerna info versioning independent

 >  Lerna (powered by Nx)   Running target build:esm for 19 projects:

    - web3-core
    - web3-errors
    - web3-eth-abi
    - web3-eth-accounts
    - web3-eth-contract
    - web3-eth-ens
    - web3-eth-iban
    - web3-eth-personal
    - web3-eth
    - web3-net
    - web3-providers-http
    - web3-providers-ipc
    - web3-providers-ws
    - web3-rpc-methods
    - web3-types
    - web3-utils
    - web3-validator
    - web3
    - web3-packagetemplate

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————— 

> web3-types:"build:esm"


> web3-packagetemplate:"build:esm"

web3-types: $ tsc --build tsconfig.esm.json && echo '{"type": "module"}' > ./lib/esm/package.json
web3-packagetemplate: $ tsc --build tsconfig.esm.json
web3-types: tsconfig.esm.json(1,1): error TS1005: '{' expected.
web3-types: tsconfig.esm.json(1,2): error TS1003: Identifier expected.
web3-types: tsconfig.esm.json(1,3): error TS1003: Identifier expected.
web3-types: tsconfig.esm.json(1,4): error TS1109: Expression expected.
web3-types: tsconfig.esm.json(1,5): error TS1003: Identifier expected.
web3-types: tsconfig.esm.json(1,6): error TS1003: Identifier expected.
web3-types: tsconfig.esm.json(1,31): error TS1005: '}' expected.
web3-types: tsconfig.esm.json(1,1): error TS1327: String literal with double quotes expected.
web3-types: tsconfig.esm.json(1,1): error TS1328: Property value can only be string literal, numeric literal, 'true', 'false', 'null', object literal or array literal.
web3-packagetemplate: tsconfig.esm.json(1,1): error TS1005: '{' expected.
web3-packagetemplate: tsconfig.esm.json(1,2): error TS1003: Identifier expected.
web3-packagetemplate: tsconfig.esm.json(1,3): error TS1003: Identifier expected.
web3-packagetemplate: tsconfig.esm.json(1,4): error TS1109: Expression expected.
web3-packagetemplate: tsconfig.esm.json(1,5): error TS1003: Identifier expected.
web3-packagetemplate: tsconfig.esm.json(1,6): error TS1003: Identifier expected.
web3-packagetemplate: tsconfig.esm.json(1,31): error TS1005: '}' expected.
web3-packagetemplate: tsconfig.esm.json(1,1): error TS1327: String literal with double quotes expected.
web3-packagetemplate: tsconfig.esm.json(1,1): error TS1328: Property value can only be string literal, numeric literal, 'true', 'false', 'null', object literal or array literal.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
web3-packagetemplate: error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————— 

 >  Lerna (powered by Nx)   Running target build:esm for 19 projects failed

   Tasks not run because their dependencies failed or --nx-bail=true:

   - web3-core:build:esm
   - web3-errors:build:esm
   - web3-eth-abi:build:esm
   - web3-eth-accounts:build:esm
   - web3-eth-contract:build:esm
   - web3-eth-ens:build:esm
   - web3-eth-iban:build:esm
   - web3-eth-personal:build:esm
   - web3-eth:build:esm
   - web3-net:build:esm
   - web3-providers-http:build:esm
   - web3-providers-ipc:build:esm
   - web3-providers-ws:build:esm
   - web3-rpc-methods:build:esm
   - web3-utils:build:esm
   - web3-validator:build:esm
   - web3:build:esm

   Failed tasks:

   - web3-types:build:esm
   - web3-packagetemplate:build:esm

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

</p>
</details>

Reply:
@jdevcs when will the update be released?  I need this feature to port the code to 4.x.

Reply:
When this PR is merged `dev` tag will be available in npm right away for your testing, and It will be part of main release after 4.1.0, next week.

Reply:
> > apologies @jdevcs are you suggesting any change?
> 
> This should be added in changelog files : https://github.com/shubhamshd/web3.js/blob/4.x/packages/web3-core/CHANGELOG.md and https://github.com/shubhamshd/web3.js/blob/4.x/packages/web3/CHANGELOG.md

Ok I just need to update the changelog files right ? under unreleased section?
I'll just create new commit and update the existing pr thanks @jdevcs

Reply:
yes, please update these changelog files

Reply:
done , please review @avkos  @jdevcs

Reply:
Due to recent release there are some file conflicts, 
could you see those, or one of our team member can fix if you dnt have time. I am approving PR but I'll merge PR after fixes. Thanks

Reply:
@jdevcs  I have resolved the conflict and updated the pr
Could you please review this now and merge, thanks!!!